### DISCUSSION TITLE
"A procedure imported by 'CefSharp.Core.Runtime.dII' could not be loaded." after upgrading old version of CefSharp.WinForms

### DISCUSSION THREAD

**Discussion by SageGary** [cefsharp/CefSharp#4411] [General]
WinForms x86 .NET Framework 4.8 app with old v57.0.0 of CefSharp.WinForms, runs OK with no issues. 
Then when I upgrade the NuGets to current v110.0.300 of CefSharp. it still compiles OK but at runtime I get
`System.IO.FileLoadException: A procedure imported by 'CefSharp.Core.Runtime.dII' could not be loaded.
`

I also compiled and ran the CefSharp.MinimalExample.WinForms.net472 project and it runs OK with no issues, so I can't be missing any dependencies? Any ideas?

Update: I already had the Visual C++ 2019 Redist installed, but I installed it again to be certain, but no change.

**Comment by amaitland** [comment]
The most common causes of this exception are:

- Attempting to load two different versions of `CEF` in the same **process**. This is a common problem when attempting to use [CefSharp as a plugin](https://github.com/cefsharp/CefSharp/discussions/4409#discussioncomment-5184326)
- Running on an [unsupported `OS` ](https://www.magpcss.org/ceforum/viewtopic.php?f=18&t=19367#p53273)
- DLL mismatch (somehow the `libcef.dll` version doesn't match the one expected by `CefSharp.Core.Runtime.dll`)

Occasionally when upgrading from a very old version the `bin/obj` directories you end up with a `DLL mismatch` and need to be manually deleted (clean checkout from source control after an upgrade where you skipped a large number of versions is generally advisable). 

> I also compiled and ran the CefSharp.MinimalExample.WinForms.net472 project and it runs OK with no issues, so I can't be missing any dependencies? Any ideas?

If the `MinimalExample` runs correctly then it's likely something specific to your project. How does your project differ from the example?

**Comment by SageGary** [comment]
It's runs on Windows 10 and there is only 1 version of CEF in the process.
The exception  is thrown on the call to 
`Cef.EnableHighDPISupport();` right at the start of the app in Main().


But I have successfully tracked down the cause by deleting out code bit-by-bit until I found the code that triggers the exception: In the project properties there is this Post-build event batch script and when it is removed, the issue goes away:
```
::remove trailing slash for robocopy
set builddir=$(TargetDir)
set builddir="%builddir:~0,-1%"

"$(SolutionDir)..\lib\robocopy.exe" "$(SolutionDir)..\lib\cef" %builddir% /s

exit 0

```

This script uses the robocopy utility to copy files from a ..\lib\cef source directory to the build output directory. But what is puzzling is the lib folder doesn't actually exist so there is no robocopy or files and its not clear to me why the batch script would trigger the issue. But anyway, removing the script resolves the issue.

**Reply by vidurasilva84** [reply]
Thanks SageGary.. 
I had the same issue and commented the following in the post-build event command (in Project Properties => Build Events => post-build event command line) and it sorted the issue...

REM ECHO COPYING Xilium.CefGlue libs
REM XCOPY $(ProjectDir)lib\CefGlue $(TargetDir) /E /Y
